metabase/metabase
# Add public action page

Epic #27626

Adds a public action page available at `/public/action/:uuid` that should be publicly accessible after an action is made public. The page should display an action form, and have simple submit success and error states.

### To Verify

1. Sign in as admin
2. Go to `/admin/settings/public-sharing` and enable public sharing
3. Create a structured model (e.g. from the sample Orders table)
4. Go to `/model/:id/detail` and click the "Actions" tab
5. Create a new action (not a basic/implicit one) with a few parameters (e.g. `UPDATE orders SET discount = {{ discount }} WHERE id = {{ order_id }}`). Important: use the parameters form on the right to specify parameter types. E.g. for an example query, both `discount` and `order_id` should be numbers
6. After creating an action, click the pencil icon on the model actions tab. Click the gear icon in the modal editor topbar and make the action public using the control on the right
7. Copy the public action URL and open it in a different browser or an incognito tab
8. Ensure you can see the action form listing parameters, and ensure the "Submit" button is disabled
9. Fill in the form, click Submit, ensure you can see the "Thank you" message
10. Use the query builder to ensure that your action had a desired effect

### Demo

https://user-images.githubusercontent.com/17258145/214897324-e3696879-1d52-4ffe-a72b-9cb53d0bf470.mp4


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27747)
